### PR TITLE
Adds playlist support

### PR DESCRIPTION
**Pull request by sabieber**
Fixes: OX-6936

### CODE REVIEW

**Review by fhaScireum — APPROVED** [review]

File: lib/cinematic.ts [hunk 1]
```diff
@@ -434,6 +434,16 @@ class Cinematic {
...
 
+        let currentVideo = this.playlist.getCurrentVideo();
+        let newSource = currentVideo.getSourcesForQuality(newQuality);
+        if (!newSource) {
+            newQuality = currentVideo.getBestAvailableQuality();
+            newSource = currentVideo.getSourcesForQuality(newQuality);
+        }
+        if (!newSource) {
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (sabieber) [reply lib/cinematic.ts:439]: Not really, the first one is to fill newSource with a fallback, the second one is a very unlikely failsafe to ensure newSource is not null below

Comment (meg1502) [inline lib/cinematic.ts:439]:
```suggestion
        if (!newSource) {
            newQuality = currentVideo.getBestAvailableQuality();
        }
        if (!newSource = currentVideo.getSourcesForQuality(newQuality);) {
            return;
        }
```

this should work too, right? maybe this is more understandable?

**Review by idlira — APPROVED** [review]

File: lib/cinematic.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fhaScireum) [reply lib/cinematic.ts:439]: ic :)

Comment (sabieber) [reply lib/cinematic.ts:439]: I would be careful with actual commit suggestions in the TypeScript repos as applying them will not update the compiled js and map files.

Comment (sabieber) [reply lib/cinematic.ts:439]: It should to the same but in my optinion assigments in if clauses can potentially be pretty dangerous as it looks like the author did the comparison wrong and actually meant `==` or `===` when coming from other languages. "Fixing" it then breaks the code.

File: lib/cinematic.ts [hunk 2]
```diff
@@ -905,3 +969,66 @@ class Cinematic {
...
+    getSourcesForQuality(quality: string): VideoQuality | null {
+        if (this.sources.length === 1) {
+            return this.sources[0];
+        }
+        for (let source of this.sources) {
+            if (source.quality === quality) {
+                return source;
+            }
+        }
+        return null;
```
Comment (meg1502) [inline lib/cinematic.ts:993]: does this really make sense to always return null if no value is found and the value if there is only one?

we already need to handle the `null` value anyway --> see `handleQualityChange` calling this method, so we might as well return the best quality in the `null` case to be consistent with the implementation for 1 source

**Pull request merged**